tests/Util/PhpServicesCreatorTest.php

```diff
@@ -0,0 +1,140 @@
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader\PhpFileLoader;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\Finder\Finder;
+
+// For testing parameters conversion
+\define('GLOBAL_CONSTANT', 'value');
+
+class PhpServicesCreatorTest extends TestCase
+{
+    // For testing parameters conversion
+    const CONSTANT_NAME = 1;
+
+    /**
+     * @dataProvider getConfigurationFiles
+     */
+    public function testConvertionYamlServices(string $yamlFile, string $phpFile, string $nameTestCase)
+    {
+        $creator = new PhpServicesCreator();
+        $this->assertSame($phpFile, $creator->convert($yamlFile));
+        /*
+         * test if default comments and configuration are written but ressources can't be loaded.
+         * the container building is tested by the testLoadResources() above.
+         */
+        if ('services_load_ressources' !== $nameTestCase) {
+            $yamlContainerBuilder = new ContainerBuilder();
+            $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source'));
+            $yamlLoader->load($nameTestCase.'.yaml');
+
+            $phpContainerBuilder = new ContainerBuilder();
+            $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/'));
+            $phpLoader->load($nameTestCase.'.php');
+
+            $this->assertTrue($yamlContainerBuilder->getParameterBag() == $phpContainerBuilder->getParameterBag());
+            $this->assertTrue($yamlContainerBuilder->getDefinitions() == $phpContainerBuilder->getDefinitions());
+            $this->assertTrue($yamlContainerBuilder->getAliases() == $phpContainerBuilder->getAliases());
+            $this->assertTrue($yamlContainerBuilder->getServiceIds() == $phpContainerBuilder->getServiceIds());
+        }
+    }
+
+    public function getConfigurationFiles()
+    {
+        $finder = Finder::create()->files()->name('*.php')
+            ->in(__DIR__.'/fixtures/YamlConvert/')
+            ->exclude(['FakeClass', 'fake_files', 'container_built_differently', 'source/fixtures_load']);
+
+        foreach ($finder as $file) {
+            $nameTestCase = str_replace('.php', '', $file->getBasename());
+
+            yield $nameTestCase => [
+                file_get_contents($file->getPath().'/source/'.$nameTestCase.'.yaml'),
+                file_get_contents($file->getPath().'/'.$file->getFilename()),
+                $nameTestCase,
+            ];
+        }
+    }
+
+    public function testParameters()
+    {
+        $source = file_get_contents(__DIR__.'/fixtures/YamlConvert/source/parameters.yaml');
+        $expectedSource = file_get_contents(__DIR__.'/fixtures/YamlConvert/parameters.php');
+
+        $creator = new PhpServicesCreator();
+
+        $this->assertSame($expectedSource, $creator->convert($source));
+    }
```

Comment:
Do we need special cases for parameters or imports? It seems that these `parameters.yaml` and `parameters.php` files should be handled by the main test above.

Reply:
Resolved in slack.